modules/system/menubar.nix

```diff
@@ -0,0 +1,52 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.system.menubar;
+
+  items = {
+    Airport         = "/System/Library/CoreServices/Menu Extras/AirPort.menu";
+    Battery         = "/System/Library/CoreServices/Menu Extras/Battery.menu";
+    Bluetooth       = "/System/Library/CoreServices/Menu Extras/Bluetooth.menu";
+    Clock           = "/System/Library/CoreServices/Menu Extras/Clock.menu";
+    Displays        = "/System/Library/CoreServices/Menu Extras/Displays.menu";
+    Eject           = "/System/Library/CoreServices/Menu Extras/Eject.menu";
+    Fax             = "/System/Library/CoreServices/Menu Extras/Fax.menu";
+    HomeSync        = "/System/Library/CoreServices/Menu Extras/HomeSync.menu";
+    Ink             = "/System/Library/CoreServices/Menu Extras/Ink.menu";
+    IrDA            = "/System/Library/CoreServices/Menu Extras/IrDA.menu";
+    Keychain        = "/Applications/Utilities/Keychain Access.app/Contents/Resources/Keychain.menu";
+    PPP             = "/System/Library/CoreServices/Menu Extras/PPP.menu";
+    PPPoE           = "/System/Library/CoreServices/Menu Extras/PPPoE.menu";
+    RemoteDesktop   = "/System/Library/CoreServices/Menu Extras/RemoteDesktop.menu";
+    ScriptMenup     = "/System/Library/CoreServices/Menu Extras/Script Menu.menu";
+    TextInput       = "/System/Library/CoreServices/Menu Extras/TextInput.menu";
+    TimeMachine     = "/System/Library/CoreServices/Menu Extras/TimeMachine.menu";
+    UniversalAccess = "/System/Library/CoreServices/Menu Extras/UniversalAccess.menu";
+    User            = "/System/Library/CoreServices/Menu Extras/User.menu";
+    VPN             = "/System/Library/CoreServices/Menu Extras/VPN.menu";
+    Volume          = "/System/Library/CoreServices/Menu Extras/Volume.menu";
+    WWAN            = "/System/Library/CoreServices/Menu Extras/WWAN.menu";
+    iChat           = "/System/Library/CoreServices/Menu Extras/iChat.menu";
+  };
+
+in {
+  options = {
+    system.menubar.items = mkOption {
+      type = types.nullOr (types.listOf (types.enum (attrNames items)));
+      default = null;
+      description = "Items to display in the menubar";
+    };
+  };
+
+  config = {
+
+    system.activationScripts.menubar.text = optionalString (!isNull cfg.items) ''
+      defaults write com.apple.systemuiserver menuExtras -array ${builtins.toString (map (k: "'" + (builtins.getAttr k items) + "'") cfg.items)}
```

Comment:
I would prefer to use the defaults module as the implementation for this.

Reply:
There's other menubar settings that can be set, not all fall under the `defaults` module, so I feel that would be more messy for end-users.

I have similar feelings with some of the existing settings in the `defaults` module; e.g. settings for "Printing" -> we now have the "show expanded dialog" option, but we could also set "automatic closing of the window when printing completed", which would then be in a different config namespace. I feel the api would be much cleaner if you would have all `printing` settings together. WDYT?

Reply:
To clarify I'm only referring to the activation scripts, not the `system.menubar` option. Modules can set options for other modules as their implementation. Something like this separates the details of defaults write from the rest of the logic.

```nix
# ...
{
  options = {
    system.menubar.items = mkOption {
      type = types.listOf (types.enum (attrValues itemPaths));
      default = [];
      example = [ "Airport" ];
    };
  };
  config = {
    system.defaults.systemuiserver.menuExtras = mkIf (cfg.items != []) (map (k: getAttr k itemPaths) cfg.items);
  };
}
```

Reply:
Does that make sense?